lerobot/common/robot_devices/cameras/opencv.py

```diff
@@ -251,45 +284,45 @@ def connect(self):
                     "To find the camera index you should use, run `python lerobot/common/robot_devices/cameras/opencv.py`."
                 )
 
-            raise OSError(f"Can't access OpenCVCamera({self.camera_index}).")
+            raise OSError(f"Can't access OpenCVCamera({camera_idx}).")
 
         # Secondly, create the camera that will be used downstream.
         # Note: For some unknown reason, calling `isOpened` blocks the camera which then
         # needs to be re-created.
-        if platform.system() == "Linux":
-            self.camera = cv2.VideoCapture(f"/dev/video{self.camera_index}")
-        else:
-            self.camera = cv2.VideoCapture(self.camera_index)
-
-        if self.fps is not None:
-            self.camera.set(cv2.CAP_PROP_FPS, self.fps)
-        if self.width is not None:
-            self.camera.set(cv2.CAP_PROP_FRAME_WIDTH, self.width)
-        if self.height is not None:
-            self.camera.set(cv2.CAP_PROP_FRAME_HEIGHT, self.height)
-
-        actual_fps = self.camera.get(cv2.CAP_PROP_FPS)
-        actual_width = self.camera.get(cv2.CAP_PROP_FRAME_WIDTH)
-        actual_height = self.camera.get(cv2.CAP_PROP_FRAME_HEIGHT)
-
-        if self.fps is not None and not math.isclose(self.fps, actual_fps, rel_tol=1e-3):
-            raise OSError(
-                f"Can't set {self.fps=} for OpenCVCamera({self.camera_index}). Actual value is {actual_fps}."
-            )
-        if self.width is not None and self.width != actual_width:
-            raise OSError(
-                f"Can't set {self.width=} for OpenCVCamera({self.camera_index}). Actual value is {actual_width}."
-            )
-        if self.height is not None and self.height != actual_height:
-            raise OSError(
-                f"Can't set {self.height=} for OpenCVCamera({self.camera_index}). Actual value is {actual_height}."
-            )
+        with self.lock:
+            self.camera = VideoCapture(camera_idx)
+
+            if self.fps is not None:
+                self.camera.set(CAP_PROP_FPS, self.fps)
+            if self.width is not None:
+                self.camera.set(CAP_PROP_FRAME_WIDTH, self.width)
+            if self.height is not None:
+                self.camera.set(CAP_PROP_FRAME_HEIGHT, self.height)
+
+            actual_fps = self.camera.get(CAP_PROP_FPS)
+            actual_width = self.camera.get(CAP_PROP_FRAME_WIDTH)
+            actual_height = self.camera.get(CAP_PROP_FRAME_HEIGHT)
+
+            # Using `math.isclose` since actual fps can be a float (e.g. 29.9 instead of 30)
```

Comment:
Nit: Below, you convert fps to an int (which makes sense to me btw) `self.fps = int(actual_fps)`, but since we do this anyways, should we just require fps to be an int so we don't need to use this `math.isclose` function?